### DISCUSSION TITLE
Allow docker.io (debian-packaged version) for supervised install

### DISCUSSION THREAD

**Discussion by mjbogusz** [home-assistant/architecture#1083] [General] [answered]
As per [ADR-0014](https://github.com/home-assistant/architecture/blob/master/adr/0014-home-assistant-supervised.md), explicitly only Docker-CE (as provided by Docker Inc.) is supported. The minimum version provided is `20.10.17`.

My proposal would be to change that to the `docker.io` package, as provided in debian's base repositories (`main` suite).

---

Rationale:
* `docker-ce` get's updated very often - this, combined with unattended-upgrades (as one should have), causes HA to break on almost every unattended-upgrade (cron or similar script detecting this case is viable but still a workaround)
* `docker-ce` is not pinned to specific version - so it will happily upgrade to a new major release with breaking changes
* `docker.io` version available in apt is, as of writing this, `20.10.24+dfsg1-1+b3`, which should fulfill the version requirement
* `docker.io` is packaged The Debian Way ™️ , meaning it doesn't repackage multiple libraries already available in the system ([discussion](https://stackoverflow.com/questions/45023363/what-is-docker-io-in-relation-to-docker-ce-and-docker-ee-now-called-mirantis-k))
* `docker.io` doesn't require any additional repos to be added to the system

Caveats and open discussion questions:
* are there any features, plugins or quirks of `docker-ce` that HA depends on that are not available in `docker.io`?
* how to handle existing installations?
* would allowing both versions be possible and viable?

---

I'm posting here as I believe this to be standards/policy-related, but if you believe this discussion is better suited for the forums, I'll happily move there.

**Comment by frenck** [comment, marked as answer]
We originally did not specify this (before we enforced the ADR). The thing is, the packaged version has bitten us quite a few times. Hence, we only supported the official Docker upstream.

You are free to run any setup you like, but we don't plan on adding support (and thus binding ourselves to package versions of a Linux Distribution) for this.

../Frenck
